core/src/main/java/hudson/model/LoadBalancer.java

```diff
@@ -86,7 +86,7 @@ public Mapping map(Task task, MappingWorksheet ws) {
 
                 // Build a Map to pass in rather than repeatedly calling hash.add() because each call does lots of expensive work
                 List<ExecutorChunk> chunks = ws.works(i).applicableExecutorChunks();
-                Map<ExecutorChunk, Integer> toAdd = Maps.newHashMapWithExpectedSize(chunks.size());
+                Map<ExecutorChunk, Integer> toAdd = new HashMap( chunks.size());
```

Comment:
This creates an empty `HashMap` with the specified initial capacity and the default load factor (0.75). If you want to avoid rehashing the map, you'll need to do something like what Guava does internally:

```java
/**
 * Returns a capacity that is sufficient to keep the map from being resized as long as it grows no
 * larger than expectedSize and the load factor is ≥ its default (0.75).
 */
static int capacity(int expectedSize) {
  if (expectedSize < 3) {
    checkNonnegative(expectedSize, "expectedSize");
    return expectedSize + 1;
  }
  if (expectedSize < Ints.MAX_POWER_OF_TWO) {
    // This is the calculation used in JDK8 to resize when a putAll
    // happens; it seems to be the most conservative calculation we
    // can make.  0.75 is the default load factor.
    return (int) ((float) expectedSize / 0.75F + 1.0F);
  }
  return Integer.MAX_VALUE; // any large value
}
```

But really the use of `Maps.newHashMapWithExpectedSize` in the first place smells like an example of [premature optimization](https://en.wikiquote.org/wiki/Donald_Knuth). So my suggestion is to just use the default default initial capacity (16) and the default load factor (0.75). Best to write the simplest and clearest code and optimize only if and when necessary.

Reply:
I don't have strong feeling here. using empty constructor (i.e default size/loadfactor) or using constructor with size will both rehash anyway :)